src/nvim/buffer.c

```diff
@@ -2833,15 +2816,16 @@ build_stl_str_hl (
       Trunc
     }               type;
   }           item[STL_MAX_ITEM];
-  int minwid;
-  int maxwid;
-  int zeropad;
-  char_u base;
-  char_u opt;
+
+  typedef enum {
+     DECIMAL     = 10,
+     OCTAL       = 8,
+     HEXIDECIMAL = 16
+  } number_base;
```

Comment:
I see, your issue was with the enum being inside the function. That is the piece that I missed. I will update.